src/kagglehub/gcs_upload.py

```diff
@@ -150,6 +156,11 @@ def _upload_blob(file_path: str, model_type: str) -> str:
                 retry_count += 1
                 uploaded_bytes = _check_uploaded_size(session_uri, file_size)
                 pbar.n = uploaded_bytes  # Update progress bar to reflect actual uploaded bytes
+            finally:
+                if file_size == 0:
+                    upload_finished = retry_count >= MAX_RETRIES
```

Comment:
This logic is wrong, if the upload of an empty file succeeds, we will be stuck in a infinite loop because `retry_count` will always be 0.

Reply:
Nevermind, we `return` on success. The logic seems all good.

Reply:
When the upload succeeds, I think it will just return as in line 144? 
But yes this is not very intuitive, wonder if there is a better way...

Reply:
I think dropping this `finally` block and using might be slightly more readable:

```
while retry_count < MAX_RETRIES and (file_size == 0 or uploaded_bytes < file_size):
```

Reply:
Yes I think this is more explicit and better. Thank you for the comments and review! Done.